lib/pac55xx/can.c

```diff
@@ -0,0 +1,470 @@
+					| ((~db2_mask << 0) & CAN_ACR_SINGLE_STD_DB2);
+	CAN_AMR(canport) = __builtin_bswap32(word);
+	/* 1: single filter */
+	CAN_ISR_SR_CMR_MR_SET(canport, CAN_MR_AFM);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief CAN Filter Single Extended Frame
+Note: Acceptance Code Mask Register values of 1 indicate the filter is to
+ignore the bit. However, standard CAN driver APIs use a positive logic for the
+mask. So this function inverts the mask to make this more portable/intuitive.
+
+@param[in] canport Unsigned int32. CAN block register base address.
+@param[in] id Unsigned int32. CAN ID. Only bits 28:0 are used.
+@param[in] id_mask Unsigned int32. CAN ID mask. Only bits 28:0 are used.
+@param[in] rtr bool. Remote Request bit value.
+@param[in] rtr_mask bool. Remote Request mask value.
+*/
+void can_filter_single_ext(uint32_t canport, uint32_t id, uint32_t id_mask) {
+	/* set value */
+	uint32_t word = ((id << 3) & CAN_ACR_SINGLE_EXT_ID);
+	CAN_ACR(canport) = __builtin_bswap32(word);
+	/* set mask */
+	word = ((~id_mask << 3) & CAN_ACR_SINGLE_EXT_ID) | CAN_ACR_SINGLE_EXT_RTR;
+	CAN_AMR(canport) = __builtin_bswap32(word);
+	/* 1: single filter */
+	CAN_ISR_SR_CMR_MR_SET(canport, CAN_MR_AFM);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief CAN Filter Single Extended Frame w/RTR set
+Note: Acceptance Code Mask Register values of 1 indicate the filter is to
+ignore the bit. However, standard CAN driver APIs use a positive logic for the
+mask. So this function inverts the mask to make this more portable/intuitive.
+
+@param[in] canport Unsigned int32. CAN block register base address.
+@param[in] id Unsigned int32. CAN ID. Only bits 28:0 are used.
+@param[in] id_mask Unsigned int32. CAN ID mask. Only bits 28:0 are used.
+@param[in] rtr bool. Remote Request bit value.
+@param[in] rtr_mask bool. Remote Request mask value.
+*/
+void can_filter_single_ext_rtr(uint32_t canport, uint32_t id, uint32_t id_mask) {
+	/* set value */
+	uint32_t word = ((id << 3) & CAN_ACR_SINGLE_EXT_ID) | CAN_ACR_SINGLE_EXT_RTR;
+	CAN_ACR(canport) = __builtin_bswap32(word);
+	/* set mask */
+	word = ((~id_mask << 3) & CAN_ACR_SINGLE_EXT_ID);
+	CAN_AMR(canport) = __builtin_bswap32(word);
+	/* 1: single filter */
+	CAN_ISR_SR_CMR_MR_SET(canport, CAN_MR_AFM);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief CAN Enable IRQ
+@param[in] canport Unsigned int32. CAN block register base address.
+@param[in] irq Unsigned int8. IRQ bit(s).
+*/
+void can_enable_irq(uint32_t canport, uint8_t imr) {
+	/* set to 1 (not masked) to enable */
+	CAN_BTR1_BTR0_RMC_IMR(canport) |= (uint32_t)imr;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief CAN Disable IRQ
+@param[in] canport Unsigned int32. CAN block register base address.
+@param[in] irq Unsigned int8. IRQ bit(s).
```

Comment:
doc parameter doesn't match parameter?

Reply:
_please_ run "make doc" and then read the doc/pac55xx/doxygen_pac55xx.log to look for warnings in the files you're adding

Reply:
Looks like I've resolved the Doxy issues.